plugins/micromega/simplex.ml

```diff
@@ -471,43 +484,19 @@ let make_farkas_proof (env : WithProof.t IMap.t) vm v =
 
 let frac_num n = n -/ Num.floor_num n
 
-(* [resolv_var v rst tbl] returns (if it exists) a restricted variable vr such that v = vr *)
-exception FoundVar of int
+type ('a, 'b) hitkind =
+  | Forget
+  (* Not interesting *)
+  | Hit of 'a
+  (* Yes, we have a positive result *)
+  | Keep of 'b
 
-let resolve_var v rst tbl =
-  let v = Vect.set v (Int 1) Vect.null in
-  try
-    IMap.iter
-      (fun k vect ->
-        if Restricted.is_restricted k rst then
-          if Vect.equal v vect then raise (FoundVar k) else ())
-      tbl;
-    None
-  with FoundVar k -> Some k
-
-let prepare_cut env rst tbl x v =
-  (* extract the unrestricted part *)
-  let unrst, rstv =
-    Vect.partition
-      (fun x vl ->
-        (not (Restricted.is_restricted x rst)) && frac_num vl <>/ Int 0)
-      (Vect.set 0 (Int 0) v)
-  in
-  if Vect.is_null unrst then Some rstv
-  else
-    Some
-      (Vect.fold
-         (fun acc k i ->
-           match resolve_var k rst tbl with
-           | None -> acc (* Should not happen *)
-           | Some v' -> Vect.set v' i acc)
-         rstv unrst)
+(* Not quite what we want but... *)
```

Comment:
Can you say more, for future reference?

Reply:
Or maybe this applies to the last constructor? In which case, we could just remove the empty line.

Reply:
Outdated comment. Let remove it entirely.